### DISCUSSION TITLE
Using EdgeX with UT

### DISCUSSION THREAD

**Discussion by idiazcir** [edgexfoundry/community#277] [Q&A] [answered]
Hello!
 
My team and I are working extensively with EdgeX Foundry. We always try to add as many UT as we can. Lately we have been developing applications on the north that utilize the SDK to send commands to the devices or get information from them. Testing with UT the functions on these applications seems quite hard as in order to test them we would need to have all of EdgeX running (to be able to use the SDK).
 
Is there any approach or suggestion for this problem?
 
Thanks in advance

**Comment by cloudxxx8** [comment, marked as answer]
To run the UT, you may use the interface mocks in SDK
https://github.com/edgexfoundry/app-functions-sdk-go/tree/main/pkg/interfaces/mocks

**Reply by idiazcir** [reply]
Nice, we will look into it. Thank you!
